src/advanced_inv.cpp

```diff
@@ -939,21 +919,26 @@ void advanced_inventory::recalc_pane( side p )
             if( s.id < AIM_SOUTHWEST || s.id > AIM_NORTHEAST ) {
                 continue;
             }
-            // deal with squares with ground + vehicle storage
-            if( s.can_store_in_vehicle() ) {
+
+            // To allow the user to transfer all items from all surrounding squares to
+            // a specific square, filter out items that are already on that square.
+            // e.g. left pane AIM_ALL, right pane AIM_NORTH. The user holds the
+            // enter key down in the left square and moves all items to the other side.
+            const bool same = other.is_same( s );
+            const bool there_veh = there.in_vehicle() && other.can_store_in_vehicle();
+
+            // Deal with squares with ground + vehicle storage
+            // Also handle the case when the other tile covers vehicle
+            // or the ground below the vehicle.
+            if( s.can_store_in_vehicle() && !( same && there_veh ) ) {
                 bool do_vehicle = ( there.get_veh_area() == s.id ) ? !there.in_vehicle() : true;
                 pane.add_items_from_area( s, do_vehicle );
                 alls.volume += s.volume;
                 alls.weight += s.weight;
-            } else {
-                // to allow the user to transfer all items from all surrounding squares to
-                // a specific square, filter out items that are already on that square.
-                // e.g. left pane AIM_ALL, right pane AIM_NORTH. The user holds the
-                // enter key down in the left square and moves all items to the other side.
-                if( other.is_same( s ) ) {
-                    continue;
-                }
-                // add map items
+            }
+
+            // Add map items
+            if( !same || there_veh ) {
```

Comment:
Something around here is wrong. My usual looting setup is dragging wheelbarrow or similar, have the left AIM panel set to ALL and the right panel to grabbed vehicle. I used to open the AIM, hold down ENTER until all items on the left had been moved to the vehicle.

Currently, it moves the items, but it also displays the items in the grabbed vehicle on the left side, which it shouldn't.

At this point, I get `same` being true, but `there_veh` also being true (where `s.id == AIM_NORTH` and `there` being AIM_DRAGGED.

Btw. trying to move the item (that is already in the grabbed vehicle) from the ALL to the vehicle does not do anything (no message, no error), that seems fine.